src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  health = 100;
+
+  constructor(name) {
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (
+      Animal.alive.includes(victim) &&
+      victim instanceof Herbivore === true &&
+      victim.health > 0 &&
+      victim.hidden === false
```

Comment:
The condition checks in the `bite` method are correct, but it's good practice to group related conditions together for readability. Consider combining the checks for `victim` being a `Herbivore` and `victim` being alive into one condition.